bitrisescript/docker.d/init_worker.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+set -o errexit -o pipefail
+
+test_var_set() {
+  local varname=$1
+
+  if [[ -z "${!varname}" ]]; then
+    echo "error: ${varname} is not set"
+    exit 1
+  fi
+}
+
+case $COT_PRODUCT in
+  mobile)
+    case $ENV in
+      dev|fake-prod)
+        test_var_set 'BITRISE_ACCESS_TOKEN_STAGING'
+        ;;
+      prod)
+        test_var_set 'BITRISE_ACCESS_TOKEN_PROD'
```

Comment:
I know we name these environment variables differently for stage/prod elsewhere, but seeing this in the context of a new script makes me wonder if we should just use `BITRISE_ACCESS_TOKEN`, and leave it to k8s to deal with setting it appropriately. I'm not sure what value the different names provide (maybe a very small amount of footgun protection?), and they certainly add complexity to this and `worker.yml`.

Reply:
Yeah, I debated this myself. I think it protects against us accidentally copying the wrong token in SOPS. If I mistakenly add `BITRISE_ACCESS_TOKEN_PROD` under the nonprod block, it'll be ignored. Whereas if the name is the same and I do that, we'll potentially have dev workers affecting production.

That said, your comment made me realize that in Bitrise, API tokens are scoped globally to the org. So in this particular case we do want both level 1 and level 3 workers to use the same token (and sensitive workflows need to be protected by scopes).

We could theoretically still use a "STAGING" token for dev workers, but this would require us to make a new org in Bitrise just for this purpose. So I think you're right and there's no point in having two separate variables.